### PR TITLE
Created Pancake Sort in C#

### LINKED ISSUE #2151: Pancake Sort
- [x] C
- [x] C++
- [x] C# - @himanshusharma89
- [ ] CoffeeScript
- [ ] Dart - @Rukmini-Meda
- [x] Go - @KamalDGRT
- [ ] Java - @parjanyaacoder
- [x] JS - @sejalshri
- [ ] Kotlin - @sandhyabhan
- [ ] Php - @sakshikhachane
- [x] Python
- [ ] Readme - @Tawishi
- [x] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by himanshusharma89**
@jainaman224 

Fixes #2151 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Pancake Sort Algorithm in C#

#### Languages Used:
- C#

#### Files Added:
- Pancake_Sort.cs

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Pancake_Sort/Pancake_Sort.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        int temp, start = 0;
+        while (start < i)
+        {
+            temp = arr[start];
+            arr[start] = arr[i];
+            arr[i] = temp;
+            start++;
+            i--;
+        }
+    }
```
Comment (asha15) [inline Pancake_Sort/Pancake_Sort.cs]: keep a blank line after this

File: Pancake_Sort/Pancake_Sort.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        for (mi = 0, i = 0; i < n; ++i)
+            if (arr[i] > arr[mi])
+                mi = i;      
+        return mi;
+    }
+ 
+    static int pancakeSort(int []arr, int n)
+    {
+        for (int curr_size = n; curr_size > 1;
+                                  --curr_size)
```
Comment (asha15) [inline Pancake_Sort/Pancake_Sort.cs]: take this to the previous line

File: Pancake_Sort/Pancake_Sort.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            if (mi != curr_size - 1)
+            {
+                flip(arr, mi);
+                flip(arr, curr_size - 1);
+            }
+        } 
+        return 0;
+    }
+ 
+    static void printArray(int []arr,int arr_size)
```
Comment (asha15) [inline Pancake_Sort/Pancake_Sort.cs]: keep space after comma

File: Pancake_Sort/Pancake_Sort.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        for (int i = 0; i < arr_size; i++)
+            Console.Write(arr[i] + " ");
+        Console.Write("");
+    }
+ 
+    public static void Main ()
+    {
+        int []arr = {23, 10, 20, 11, 12, 6, 7};
```
Comment (asha15) [inline Pancake_Sort/Pancake_Sort.cs]: take dynamic input

**Comment by himanshusharma89** [comment]
@asha15 please review it now.

File: Pancake_Sort/Pancake_Sort.cs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                mi = i;      
+        return mi;
+    }
+ 
+    static int pancakeSort(int []arr, int n)
+    {
+        for (int curr_size = n; curr_size > 1;--curr_size)
+        { 
+            int mi = findMax(arr, curr_size);// function calling for finding index of max integer till the index curr_size
+            if (mi != curr_size - 1)// call function flip for swapping if index of max integer till the index curr_size is not equal to curr_size-1
```
Comment (asha15) [inline Pancake_Sort/Pancake_Sort.cs]: could you take the comment to the previous line

File: Pancake_Sort/Pancake_Sort.cs [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        {
+            temp = arr[start];
+            arr[start] = arr[i];
+            arr[i] = temp;
+            start++;
+            i--;
+        }
+    }
+ 
+    static int findMax(int []arr, int n)// function returning index of max value till index curr_size
```
Comment (asha15) [inline Pancake_Sort/Pancake_Sort.cs]: take the comment to the previous line

File: Pancake_Sort/Pancake_Sort.cs [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        int mi, i;
+        for (mi = 0, i = 0; i < n; ++i)
+            if (arr[i] > arr[mi])
+                mi = i;      
+        return mi;
+    }
+ 
+    static int pancakeSort(int []arr, int n)
+    {
+        for (int curr_size = n; curr_size > 1;--curr_size)
```
Comment (asha15) [inline Pancake_Sort/Pancake_Sort.cs]: please keep space before operator

File: Pancake_Sort/Pancake_Sort.cs [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+            if (mi != curr_size - 1)// call function flip for swapping if index of max integer till the index curr_size is not equal to curr_size-1
+            {
+                flip(arr, mi);
+                flip(arr, curr_size - 1);
+            }
+        } 
+        return 0;
+    }
+ 
+    static void printArray(int []arr, int arr_size)// function for printing sorted array
```
Comment (asha15) [inline Pancake_Sort/Pancake_Sort.cs]: take the comment to the previous line

File: Pancake_Sort/Pancake_Sort.cs [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        for (int i = 0; i < arr_size; i++)
+            Console.Write(arr[i] + " ");
+        Console.Write("");
+    }
+ 
+    public static void Main (string [] args)
+    {
+        int n=Convert.ToInt32(Console.ReadLine());//input size of array
+        int []arr=new int[n];//initializing array of size n
```
Comment (asha15) [inline Pancake_Sort/Pancake_Sort.cs]: keep a line space

File: Pancake_Sort/Pancake_Sort.cs [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+        for(int i=0;i<n;i++)
+            arr[i] = Convert.ToInt32(Console.ReadLine());//input value in array
```
Comment (asha15) [inline Pancake_Sort/Pancake_Sort.cs]: keep a line space

File: Pancake_Sort/Pancake_Sort.cs [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+        for (int i = 0; i < arr_size; i++)
+            Console.Write(arr[i] + " ");
+        Console.Write("");
+    }
+ 
+    public static void Main (string [] args)
+    {
+        int n=Convert.ToInt32(Console.ReadLine());//input size of array
+        int []arr=new int[n];//initializing array of size n
+        for(int i=0;i<n;i++)
```
Comment (asha15) [inline Pancake_Sort/Pancake_Sort.cs]: keep space around operator

**Comment by asha15** [comment]
please squash the commits and push it

**Comment by himanshusharma89** [comment]
@asha15 I am facing issue with squashing the commits as I forgot to clone my forked repo and there was no pancake branch after cloning. It was even with master branch on doing checkout pancake. So can you help me with this?

**Comment by asha15** [comment]
sure. reach me on slack

File: Pancake_Sort/Pancake_Sort.cs [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+                mi = i;      
+        return mi;
+    }
+ 
+    static int pancakeSort(int []arr, int n)
+    {
+        for (int curr_size = n; curr_size > 1; --curr_size)
+        { 
+            int mi = findMax(arr, curr_size);// function calling for finding index of max integer till the index curr_size
+            // call function flip for swapping if index of max integer till the index curr_size is not equal to curr_size-1
```
Comment (asha15) [inline Pancake_Sort/Pancake_Sort.cs]: one line should contain only 80 characters. break the lines according to it

**Pull request merged**